src/ui/CommitList.cpp

```diff
@@ -809,82 +814,160 @@ class CommitDelegate : public QStyledItemDelegate
     // Adjust margins.
     rect.setY(rect.y() + kVerticalMargin);
     rect.setX(rect.x() + kHorizontalMargin);
-    rect.setWidth(rect.width() - kHorizontalMargin);
+    if (!compactMode) rect.setWidth(rect.width() - kHorizontalMargin); // Star has enough padding in compact mode
 
     // Draw content.
     git::Commit commit = index.data(CommitRole).value<git::Commit>();
     if (commit.isValid()) {
-      // Draw Name.
-      QString name = commit.author().name();
-      painter->save();
-      QFont bold = opt.font;
-      bold.setBold(true);
-      painter->setFont(bold);
-      painter->drawText(rect, Qt::AlignLeft, name);
-      painter->restore();
-
-      // Draw date.
       const QFontMetrics &fm = opt.fontMetrics;
-      QDateTime date = commit.committer().date().toLocalTime();
-      QString timestamp =
-        (date.date() == QDate::currentDate()) ?
-        date.time().toString(Qt::DefaultLocaleShortDate) :
-        date.date().toString(Qt::DefaultLocaleShortDate);
-      if (rect.width() > fm.width(name) + fm.width(timestamp) + 8) {
+      QRect star = rect;
+
+      if (compactMode) {
+        int maxWidthRefs = (int)(rect.width() * 0.5); // Max 30% of the 
+        const int minWidthRefs = 50; // At least display The ellipsis
+        const int minWidthRequestDesc = 100;
+        int minDisplayWidthDate = 350;
+        int minDisplayWidthName = 800;
+        int minWidthName = 50;
+        int maxWidthName = (int)(rect.width() * 0.15);
+
+        // Star always takes up its height on the right side
+        star.setX(star.x() + star.width() - star.height());
+        star.setY(star.y() - kVerticalMargin);
+
+        // maybe..maybe?? not needed in compact mode?
+        // can't really remove it, at least it 
+        // needs to be an option
+        QString id = commit.shortId();
+        QRect box = rect;
+        box.setWidth(box.width() - star.width());
+        // Using the biggest theoretical width
+        int idWidth = fm.horizontalAdvance("9999999") + kHorizontalMargin;
+        painter->save();
+        painter->drawText(box, Qt::AlignRight, id);
+        painter->restore();
+        box.setWidth(box.width() - idWidth);
+
+        // Draw date. Only if String is not the same as previous?
+        QDateTime date = commit.committer().date().toLocalTime();
+        QString timestamp =
+          (date.date() == QDate::currentDate()) ?
+          date.time().toString(Qt::DefaultLocaleShortDate) :
+          date.date().toString(Qt::DefaultLocaleShortDate);
+        if (box.width() > minWidthRequestDesc + fm.horizontalAdvance(timestamp) + 8 && totalWidth > minDisplayWidthDate) {
```

Comment:
I intend to add a check with previous string != current string here, but I don't know where to store the previous string